### ISSUE TITLE
unknown lhs 86

### ISSUE THREAD

**Issue by zzak** [mruby/mruby#2781]
I have the following Ruby code:

```ruby
def complete(key, icase = false, pat = nil)
  candidates = candidate(key, icase, pat, &method(:each)).sort_by {|k, v, kn| kn.size}
  if candidates.size == 1
    canon, sw, * = candidates[0]
  end
end
```

It seems this line won't compile in mruby, and generates a `unknown lhs 86` in `gem_init.c` when compiling this code in an mrbgem.

`canon, sw, * = candidates[0]`

However, if I change it to this it works:

`canon, sw, _ = candidates[0]`

Is this a bug in mruby parser?

**Comment by cremno** [comment]
I can't reproduce it (current HEAD / 2fe556d9c039839c20965a2c90dff703f04e40ec). It was most likely fixed by #2684.

**Comment by zzak** [comment]
Ahh, maybe! I'm using the stable version, let me try again with master.

Thanks!

**Comment by tsahara** [comment]
I can reproduce this on latest mruby 2fe556d9c039839c20965a2c90dff703f04e40ec built for OpenBSD/amd64 or MacOS.

Here is a shorter code:

```ruby
a, * = [1, 2]
p a
```

```
% ruby a.rb
1

% mruby a.rb
unknown lhs 86
1
```

It shows `unknown lhs 84` (not `86`) on MacOS though.

**Comment by cremno** [comment]
@tsahara: Oh, I actually can reproduce it too. But I don't know why it's `84`/`NODE_ATTRSET` and not `86`/`NODE_NIL` on MacOS. Sorry, @zzak! I've created a PR.

**Comment by tsahara** [comment]
> But I don't know why it's 84/NODE_ATTRSET...

Sorry, it's my mistake.  It shows `unknown lhs 86` when I tested again.
